src/structarray.jl

```diff
@@ -445,7 +445,11 @@ end
 # broadcast
 import Base.Broadcast: BroadcastStyle, ArrayStyle, AbstractArrayStyle, Broadcasted, DefaultArrayStyle
 
-struct StructArrayStyle{Style} <: AbstractArrayStyle{Any} end
+struct StructArrayStyle{S,N} <: AbstractArrayStyle{N} end
+# If `S` also track input's dimensionality, we'd better also update it.
```

Comment:
I confess I'm not fully familiar with the broadcast machinery, but I imagine this comment could be either expanded or removed, as it is it is a bit cryptic (it only makes sense if one knows the diff from this PR).